samples/nrf9160/nrf_cloud_rest_device_message/README.rst

```diff
@@ -0,0 +1,92 @@
+
+Requirements
+************
+
+The sample supports the following development kits:
+
+.. table-from-rows:: /includes/sample_board_rows.txt
+   :header: heading
+   :rows: nrf9160dk_nrf9160_ns
+
+The sample requires an `nRF Cloud`_ account.
+
+.. note::
+   This sample requires modem firmware v1.3.x or later.
+
+Overview
+********
+
+Pressing the button sends a buttonpress message that can be viewed in the nRF Cloud terminal. The device must already be provisioned. However, you can provision the device over REST by enabling :kconfig:`CONFIG_REST_DEVICE_MESSAGE_DO_JITP` and following the instructions printed to UART upon boot.
+
+Limitations
+***********
+
+This sample requires the network carrier to provide date and time to the modem.
+Without a valid date and time, the modem cannot generate JWTs with an expiration time.
+
+User interface
+**************
+
+Once provisioned and connected, each press of the configured button (configured using the :kconfig:`CONFIG_REST_DEVICE_MESSAGE_BUTTON_EVT_NUM` option) generates a device to cloud buttonpress device message over REST.
+
+If you have the option :kconfig:`CONFIG_REST_DEVICE_MESSAGE_DO_JITP` enabled and you press the button when prompted at startup, it will request just-in-time provisioning (JITP) through REST with nRF Cloud.
+This is useful when initially provisioning and associating a device on nRF Cloud.
+You only need to do this once for each device.
+
+The configured LED (:kconfig:`CONFIG_REST_DEVICE_MESSAGE_LED_NUM`) will light once an LTE connection is established and JWT tokens are ready to be generated. Assuming the device is provisioned, this is analogous to being connected to nRF Cloud, though the REST API does not truly employ connections beyond the transient connections inherent to HTTP requests.
+
+Configuration
+*************
+|config|
+
+Configuration options
+=====================
+
+Check and configure the following configuration options for the sample:
+
+.. kconfig:: CONFIG_REST_DEVICE_MESSAGE_LED_NUM
+
+CONFIG_REST_DEVICE_MESSAGE_LED_NUM - Enable LED
+   This configuration option defines if the LED is to be used.
+
+.. kconfig:: CONFIG_REST_DEVICE_MESSAGE_BUTTON_EVT_NUM
+
+CONFIG_REST_DEVICE_MESSAGE_BUTTON_EVT_NUM - Button number
+   This configuration option defines the button to use for device interactions.
+
+.. kconfig:: CONFIG_REST_DEVICE_MESSAGE_DO_JITP
+
+CONFIG_REST_DEVICE_MESSAGE_DO_JITP - Enable prompt to perform JITP via REST
+   This configuration option defines if the application will prompt the user for just-in-time provisioning on startup.
+
+Building and running
+********************
+
+.. |sample path| replace:: :file:`samples/nrf9160/nrf_cloud_rest_device_message`
+
+.. include:: /includes/build_and_run_nrf9160.txt
+
+The configuration file for this sample is located in :file:`samples/nrf9160/nrf_cloud_rest_device_message`.
+See :ref:`configure_application` for information on how to configure the parameters.
```

Comment:
Do these messages show up in the UI on nRF Cloud?

If so, is there a limitation on this? For example, the device had to have already been provisioned for MQTT access.

If not, how does the customer verify that the messages arrive at nRF Cloud?  For example, is there a CURL command they could issue to retrieve historic data?

Reply:
The button messages and hello world message appear on the front end, yep!

Yes, the device must already be provisioned. JITP can be requested if it was not previously provisioned

I will try to make that clearer in documentation

Reply:
You can put a link from README.rst to the nRF Cloud docs, through a doc symbol defined in sdk-nrf/doc/nrf/links.txt.  There is a section of links there under the heading, where you could add one that you need:
```
... ### Source: nrfcloud.com

Reply:
Sure, are you wanting me to dig up some documentation on accessing device messages via CURL and include that? We already link to Device Message documentation and to nrfcloud.com

Reply:
Yes, thanks.  Let me know if you run into trouble.  It's a nice-to-have, for customers that register a device for REST access but not nRF Cloud MQTT access.